crates/gitbutler-core/src/virtual_branches/branch/hunk.rs

```diff
@@ -17,7 +20,7 @@ impl From<&diff::GitHunk> for Hunk {
         Hunk {
             start: hunk.new_start,
             end: hunk.new_start + hunk.new_lines,
-            hash: Some(Hunk::hash(&hunk.diff)),
+            hash: Some(Hunk::hash(hunk.diff_lines.as_ref())),
```

Comment:
Here a `hash` that is supposed to be created by `Hunk::hash_diff` is set with one that is just a normal hash. That *should* be fine as `diff_lines` are only lines, without a header, BUT it's not the same as the line-endings are included here. Even if `hash_diff()` would include line endings as well, previously [these have been synthetic](https://github.com/gitbutlerapp/gitbutler/blob/3005c067298edd7c6670fbec99f36d8b778296f2/crates/gitbutler-core/src/virtual_branches/branch/hunk.rs#L165) as well.
So no matter what, I don't think these hashes will be very comparable unless great care is taken regarding the way these lines are tokenized, and how that token is (or isn't) reproduced in the hash.

Reply:
Hmm, after fiddling with it more it seems that possible values for `diff_lines` aren't as diverse as I thought and most of them do contains the `@@` header. Probably this hasn't been a cause of problems.